src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.io.BufferedReader;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader br = Files.newBufferedReader(Path.of(fromFileName))) {
+            String line;
+
+            while ((line = br.readLine()) != null) {
+                String[] values = line.split(",");
+                if (values.length == 2 && values[0].equals("supply")) {
+                    totalSupply += Integer.parseInt(values[1]);
+                } else if (values.length == 2 && values[0].equals("buy")) {
+                    totalBuy += Integer.parseInt(values[1]);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read this file", e);
```

The catch block correctly throws a `RuntimeException` with the original exception and a message. This aligns with the checklist item: "Don't ignore exceptions: Better re-throw `RuntimeException` with the original exception and some message in the parameters."